grunt/config/watch.js

```diff
@@ -33,62 +33,66 @@ module.exports = {
             '!<%= sourcedir %>theme/theme.js',
             '!<%= sourcedir %>templates/templates.js',
             '!<%= sourcedir %>core/js/scriptLoader.js',
-            '!<%= sourcedir %>core/js/libraries/require.min.js',
-            '!<%= sourcedir %>core/js/libraries/modernizr.js',
-            '!<%= sourcedir %>core/js/libraries/jquery.min.js'
+            '!<%= sourcedir %>core/js/libraries/**/*.js',
+            '!<%= sourcedir %>extensions/*/libraries/**/*.js',
+            '!<%= sourcedir %>components/**/libraries/**/*.js',
+            '!<%= sourcedir %>menu/<%= menu %>/libraries/**/*.js', 
+            '!<%= sourcedir %>theme/<%= theme %>/libraries/**/*.js'
         ],
         tasks: ['javascript:dev']
     },
     index: {
         files: ['<%= sourcedir %>index.html'],
-        tasks: ['copy:index']
+        tasks: ['newer:copy:index']
     },
     componentsAssets: {
         files: ['<%= sourcedir %>components/**/assets/**'],
-        tasks: ['copy:componentAssets']
+        tasks: ['newer:copy:componentAssets']
     },
     componentsFonts: {
         files: ['<%= sourcedir %>components/**/fonts/**'],
-        tasks: ['copy:componentFonts']
+        tasks: ['newer:copy:componentFonts']
     },
     extensionsAssets: {
         files: ['<%= sourcedir %>extensions/**/assets/**'],
-        tasks: ['copy:extensionAssets']
+        tasks: ['newer:copy:extensionAssets']
     },
     extensionsFonts: {
         files: ['<%= sourcedir %>extensions/**/fonts/**'],
-        tasks: ['copy:extensionFonts']
+        tasks: ['newer:copy:extensionFonts']
     },
     menuAssets: {
         files: ['<%= sourcedir %>menu/<%= menu %>/**/assets/**'],
-        tasks: ['copy:menuAssets']
+        tasks: ['newer:copy:menuAssets']
     },
     menuFonts: {
         files: ['<%= sourcedir %>menu/<%= menu %>/**/fonts/**'],
-        tasks: ['copy:menuFonts']
+        tasks: ['newer:copy:menuFonts']
     },
     themeAssets: {
         files: ['<%= sourcedir %>theme/<%= theme %>/**/assets/**'],
-        tasks: ['copy:themeAssets']
+        tasks: ['newer:copy:themeAssets']
     },
     themeFonts: {
         files: ['<%= sourcedir %>theme/<%= theme %>/**/fonts/**'],
-        tasks: ['copy:themeFonts']
+        tasks: ['newer:copy:themeFonts']
     },
     scriptLoader: {
         files: ['<%= sourcedir %>core/js/scriptLoader.js'],
-        tasks: ['copy:scriptLoader']
+        tasks: ['newer:copy:scriptLoader']
     },
     libraries: {
-        files: [
-            '<%= sourcedir %>core/js/libraries/require.min.js',
-            '<%= sourcedir %>core/js/libraries/modernizr.js',
-            '<%= sourcedir %>core/js/libraries/jquery.min.js'
-        ],
-        tasks: ['copy:libraries']
+        files: ['<%= sourcedir %>core/js/libraries/**/*'],
```

Comment:
this bit should include the `libraries` folder directives for extensions, menus, components etc

Reply:
I have included the reference of the directive  for extensions, menus, components into required directives, here is the reference,

https://github.com/adaptlearning/adapt_framework/blob/b6988c335cd9f8b077c7d9f52fc379a5ddc6721b/grunt/config/copy.js#L211

Reply:
the `required` folder and the `libraries` folder are two different things.

Reply:
Do we really need 

> 

{
                    expand: true,
                    src: ['components/**/libraries/**/*', 'extensions/**/libraries/**/*', 'menu/<%= menu %>/libraries/**/*', 'theme/<%= theme %>/libraries/**/*'],
                    cwd: '<%= sourcedir %>',
                    dest: '<%= outputdir %>/libraries/',
                    filter: function(filepath) {
                        return grunt.config('helpers').includedFilter(filepath);
                    },
                    rename: _.partial(collate, "libraries")
                }

> at  https://github.com/adaptlearning/adapt_framework/blob/b6988c335cd9f8b077c7d9f52fc379a5ddc6721b/grunt/config/copy.js#L212

so can we move this to 
https://github.com/adaptlearning/adapt_framework/blob/b6988c335cd9f8b077c7d9f52fc379a5ddc6721b/grunt/config/copy.js#L199

Reply:
Yup, that works.

Reply:
yes, I will do the change an update it soon.